File: 1.模法推荐队-可运行项目成果/frontend/src/types/inference.ts
export interface ModelInvokeRequest {
  input_data?: Record<string, unknown>;
  async_mode?: boolean;
  request_context?: Record<string, unknown>;
}

export interface ModelInvokeResponse {
  model_id: string;
  task_id: string;
  status: string;
  demo_data: boolean;
  submitted_at: string;
  message: string;
  result: ModelResultPayload;
}

export interface ModelResultResponse {
  task_id: string;
  status: string;
  demo_data: boolean;
  result: ModelResultPayload;
}

export interface ModelResultPayload {
  model_id?: string;
  demo_data?: boolean;
  result_type?: string;
  input_echo_keys?: string[];
  desensitized_notice?: string;
  compliance_notice?: string;
  usage_boundary?: string;
  compliance?: {
    result_type?: string;
    sensitivity_level?: string;
    usage_boundary?: string;
    allowed_usage?: string[];
    prohibited_usage?: string[];
    default_desensitized?: boolean;
    sensitive_fields_masked?: string[];
    field_registry_loaded?: boolean;
  };
  rows?: Record<string, unknown>[];
}

File: 1.模法推荐队-可运行项目成果/backend/app/schemas/inference.py
"""Schemas for model invocation and task/result retrieval."""

from __future__ import annotations

from typing import Any

from pydantic import BaseModel, Field


class ModelInvokeRequest(BaseModel):
    """Request payload for invoking a model."""

    input_data: dict[str, Any] = Field(default_factory=dict)
    async_mode: bool = False
    request_context: dict[str, Any] = Field(default_factory=dict)


class ModelInvokeResponse(BaseModel):
    """Response for model invocation."""

    model_id: str = ""
    task_id: str = ""
    status: str = ""
    demo_data: bool = False
    submitted_at: str = ""
    message: str = ""
    result: dict[str, Any] = Field(default_factory=dict)


class ModelTaskStatusResponse(BaseModel):
    """Async model task status."""

    task_id: str = ""
    status: str = ""
    demo_data: bool = False
    updated_at: str = ""
    message: str = ""


class ModelResultResponse(BaseModel):
    """Async model task result."""

    task_id: str = ""
    status: str = ""
    demo_data: bool = False
    result: dict[str, Any] = Field(default_factory=dict)


class ModelResultSchemaResponse(BaseModel):
    """Model result schema response."""

    model_id: str = ""
    demo_data: bool = False
    result_schema: dict[str, Any] = Field(default_factory=dict)
